src/components/GameTypes/GameTypes.js

```diff
@@ -0,0 +1,45 @@
+import { html } from '../../shared';
+import { GameTypeButton } from '../GameTypeButton';
+import styles from './GameTypes.module.css';
+
+/**
+ * @param {{onSelect: (e: MouseEvent) => void, heading: string, categories: object[], layout?: 'default' | 'halfWidth' }} props
+ */
+
+function GameTypes({ onSelect, heading, categories, layout = 'default' }) {
+  const typeStyles = {
+    default: styles.containerFullWidth,
+    halfWidth: styles.containerHalfWidth,
+  };
+
+  const selectedTypes = (e) => {
+    const container = e.target.parentElement;
+    const containerButtons = container.querySelectorAll('button');
+
+    if (e.target.matches('button')) {
+      [...containerButtons].forEach((button) => {
+        button.classList.remove(styles.selectedType);
+      });
+      e.target.classList.add(styles.selectedType);
+      onSelect(e.target);
```

Comment:
Fajnie jakby w `onSelect` było przekazywane bezpośrednio `id` kategorii/poziomu, bo to będzie na pewno potrzebne rodzicowi, a informacja, który przycisk tą kategorię zmienił już mniej. Wtedy, nawiązując do drugiego komentarza, u góry byłoby `onSelect: (id: string) => void`.

Chociaż teraz też jest w sumie ok, bo można wyciągnąć to co chcemy z `e.id` (tylko trzeba uważać żeby to `id` się nigdzie w drzewie nie powtórzyło).

Reply:
To ja bym chyba tutaj to zostawiła jak było, a ewentualnie jeśli zobaczymy jakieś problemy z tym na kolejnych etapach to będzie można to szybko zmienić.